File: backend/src/main/java/com/fretcorridor/backend/shipment/entity/PackageCatalogItem.java
package com.fretcorridor.backend.shipment.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.UUID;

/**
 * Entree du catalogue reduit (~15 items, plan §1). Lecture seule cote API
 * (pas de CRUD chargeur dessus) - alimente uniquement par la migration V4.
 */
@Entity
@Table(name = "package_catalog_items")
@Getter
@Setter
@NoArgsConstructor
public class PackageCatalogItem {

    @Id
    @GeneratedValue
    private UUID id;

    @Column(nullable = false, unique = true, length = 50)
    private String code;

    @Column(nullable = false, length = 100)
    private String label;

    @Column(nullable = false, length = 50)
    private String category;

    @Column(name = "default_weight_kg", nullable = false, precision = 8, scale = 2)
    private BigDecimal defaultWeightKg;

    @Column(name = "default_volume_m3", nullable = false, precision = 8, scale = 3)
    private BigDecimal defaultVolumeM3;

    @Column(name = "fragile_by_default", nullable = false)
    private boolean fragileByDefault;

    @Column(name = "icon_name", nullable = false, length = 50)
    private String iconName;

    @Column(nullable = false)
    private boolean active;
}

File: mobile/lib/features/shipment_request/data/models/package_catalog_item.dart
/// Entree du catalogue reduit (~15 items, cf. backend V4). Correspond 1:1
/// au DTO PackageCatalogItemResponse - aucune transformation cote client.
class PackageCatalogItem {
  PackageCatalogItem({
    required this.id,
    required this.code,
    required this.label,
    required this.category,
    required this.defaultWeightKg,
    required this.defaultVolumeM3,
    required this.fragileByDefault,
    required this.iconName,
  });

  final String id;
  final String code;
  final String label;
  final String category;
  final double defaultWeightKg;
  final double defaultVolumeM3;
  final bool fragileByDefault;
  final String iconName;

  factory PackageCatalogItem.fromJson(Map<String, dynamic> json) => PackageCatalogItem(
        id: json['id'],
        code: json['code'],
        label: json['label'],
        category: json['category'],
        defaultWeightKg: (json['defaultWeightKg'] as num).toDouble(),
        defaultVolumeM3: (json['defaultVolumeM3'] as num).toDouble(),
        fragileByDefault: json['fragileByDefault'],
        iconName: json['iconName'],
      );
}
